libbpfgo.go

```diff
@@ -752,25 +752,22 @@ func (p *BPFProg) SetTracepoint() error {
 	return nil
 }
 
-func (p *BPFProg) AttachTracepoint(tp string) (*BPFLink, error) {
-	tpEvent := strings.Split(tp, ":")
-	if len(tpEvent) != 2 {
-		return nil, fmt.Errorf("tracepoint must be in 'category:name' format")
-	}
-	tpCategory := C.CString(tpEvent[0])
-	tpName := C.CString(tpEvent[1])
+func (p *BPFProg) AttachTracepoint(category, name string) (*BPFLink, error) {
+	tpCategory := C.CString(category)
+	defer C.free(unsafe.Pointer(tpCategory))
+	tpName := C.CString(name)
+	defer C.free(unsafe.Pointer(tpName))
+
 	link := C.bpf_program__attach_tracepoint(p.prog, tpCategory, tpName)
-	C.free(unsafe.Pointer(tpCategory))
-	C.free(unsafe.Pointer(tpName))
```

Comment:
nit: what was wrong with freeing the C strings here that we need to defer it?

Reply:
It's just a nitpicking, perhaps trying to use the Golang idioms. But this change is not mandatory. If rejected I can get rid of it.

Reply:
Well in this case, we can just free the allocated objects right after we use them, and it seems to be clearer (at least for me).
I would use defer when there is an operation (like closing a file) that can't happen right away, and I want to ensure that will happen in any return path of the function.